server/src/main/java/io/deephaven/server/barrage/BarrageMessageProducer.java

```diff
@@ -110,13 +115,18 @@
      * @param <MessageView> The sub-view type that the listener expects to receive.
      */
     public interface StreamGenerator<MessageView> extends SafeCloseable {
+        interface WriteMetricsConsumer {
+            void onWrite(long bits, long cpuTime);
```

Comment:
Is `bits` really a better unit than `bytes`? Feels a "bit" unusual, and it's not like we can send units smaller than bytes.